src/berry_mill/mill.py

```diff
@@ -78,6 +98,23 @@ def __init__(self):
                 if pth.split('.')[-1] in ["kiwi", "xml"]:
                     self._appliance_descr = pth
                     break
+        
+        if not self._appliance_descr:
+            raise Exception("Appliance description was not found.")
+        
+        if self._appliance_path:
+            os.chdir(self._appliance_path)
+        else:
+            raise Exception("Appliance Path not found")
```

Comment:
I would raise or chdir instead:
```python
if not self._appliance_path:
    raise Exception("AAAA!")
os.chdir(self._appliance_path)
```

Less complex to comprehend and saves one code line.